src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -145,6 +146,26 @@ public static ulong RoundUpToPowerOf2(ulong value)
             return value + 1;
         }
 
+        /// <summary>
+        /// Round the given integral value up to a power of 2.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <returns>
+        /// The smallest power of 2 which is greater than or equal to <paramref name="value"/>.
+        /// If <paramref name="value"/> is 0 or the result overflows, returns 0.
+        /// </returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CLSCompliant(false)]
+        public static nuint RoundUpToPowerOf2(nuint value)
+        {
+            if (Environment.Is64BitProcess)
```

Comment:
In situations like this one, we often use `#if TARGET_64BIT` in CoreLib to make the IL smaller and easier for JIT to inline.

Reply:
Hello @jkotas,

have now replaced the check with the preprocessor variable, which hopefully inlines the JIT better.

Best regards
Robin